ballista/rust/core/src/execution_plans/shuffle_reader.rs

```diff
@@ -156,6 +158,38 @@ impl ExecutionPlan for ShuffleReaderExec {
     fn metrics(&self) -> Option<MetricsSet> {
         Some(self.metrics.clone_inner())
     }
+
+    fn statistics(&self) -> Statistics {
+        stats_for_partitions(
```

Comment:
Perhaps this aggregated stats should be calculated once and cached as a struct field? Otherwise the same calculation needs to be done again for every optimizer right?

Reply:
I would have done it in the constructor if this object was immutable, but the `partition` field is actually `pub(crate)`. I'll open a separate PR to make `partition` and `schema` private, thus making it possible to cache the statistics.

Reply:
Opened #987 to track this